code/datums/components/payment.dm

```diff
@@ -76,3 +86,74 @@
 	if(!isnum(new_cost))
 		CRASH("change_cost called with variable new_cost as not a number.")
 	cost = new_cost
+
+/datum/component/payment/proc/handle_cardless(mob/living/user, total_cost)
+	//Here is all the possible non-ID payment methods.
+	var/list/counted_money = list()
+	var/physical_cash_total = 0
+	for(var/obj/item/coin/counted_coin in user.get_all_contents()) //Coins.
+		if(physical_cash_total > total_cost)
+			break
+		physical_cash_total += counted_coin.value
+		counted_money += counted_coin
+
+	for(var/obj/item/stack/spacecash/counted_cash in user.get_all_contents()) //Paper Cash
+		if(physical_cash_total > total_cost)
+			break
+		physical_cash_total += counted_cash.value * counted_cash.amount
+		counted_money += counted_cash
+
+	for(var/obj/item/holochip/counted_credits in user.get_all_contents()) //Holocredits
+		if(physical_cash_total > total_cost)
+			break
+		physical_cash_total += counted_credits.credits
+		counted_money += counted_credits
```

Comment:
Calling this proc 3 times is redundant. Instead:

Create a static typecache of typepaths you're ok with using
use `typecache_filter_list` to filter the list get_all_contents generates with it
Move the "get value" stuff to a separate proc
```
var/value = get_value(thing)
if(!value)
    continue
add the value and item to the deletion list
```